### ISSUE TITLE
AccessibilityRole="radio" is not translated on iOS Voiceover

### ISSUE THREAD

**Issue by Waltari10** [facebook/react-native#31227]
## Description
iOS VoiceOver screenReader does not translate `accessibilityRole="radio"` to voiceOvers selected language. Other accessibilityRole values like "button" or "header" get translated.

## React Native version:
System:
    OS: macOS 11.2.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 1012.44 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.19.0 - ~/.nvm/versions/node/v12.19.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v12.19.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6626763
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 12.0.2 - /Library/Java/JavaVirtualMachines/jdk-12.0.2.jdk/Contents/Home/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.4 => 0.63.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

## Steps To Reproduce
1. Add element e.g. TouchableOpacity
2. Give TouchableOpacity accessibilityRole="radio"
3. Change VoiceOver language to Finnish or another non-english language
4. Enable screenreader
5. Have screenreader announce TouchableOpacity
6. TouchableOpacity is announced as "Radio button" instead of translated equilevant

## Expected Results
Expected VoiceOver screenreader announcement "Radio button" to be translated to VoiceOvers selected language e.g. Finnish in this case, just like with other "accessibilityRole" prop values like "button" or "header".

## Snack, code example, screenshot, or link to a repository:
example repo: https://github.com/Waltari10/iosReactNativeRadio/blob/master/App.js 

import React from 'react';
import {SafeAreaView, Text, StatusBar, TouchableOpacity} from 'react-native';

// No translation exists in finnish for radio button with voiceOver

```
const App = () => {
  return (
    <>
      <StatusBar barStyle="light-content" />
      <SafeAreaView style={{margin: 40}}>
        <TouchableOpacity accessibilityRole="radio">
          {/* If finnish VoiceOver language set, should be read out as finnish equilevant "valintanappi", instead read out as "radio button" */}
          <Text style={{color: 'white'}}>Minun radio nappini</Text>
        </TouchableOpacity>
      </SafeAreaView>
    </>
  );
};

export default App;
```

**Comment by Waltari10** [comment]
Same thing with accessibilityRole="radiogroup"

**Comment by Waltari10** [comment]
I noticed that the "radio button" string was defined in `vr-mobile-ng/node_modules/react-native/React/AccessibilityResources/en.lproj/Localizable.strings`

Overwriting it there directly seemed to reflect in the app, but I wonder how to overwrite it without modifying the node_modules directly 🤔

**Comment by Waltari10** [comment]
Wrote an article about the hacky solution I implemented: https://valtterilaine.medium.com/react-native-accessibilityrole-i18n-support-for-ios-and-android-16d896e69178

**Comment by kacieb** [comment]
Thanks for reporting this! I didn't realize this was happening on iOS as well (I know this is a problem on Android due to some hardcoded English strings). I will take a look.

**Comment by veraldo** [comment]
I’m not an iOS developer so somebody correct me if I’m wrong, but it seems like this “radio” role does not exist on native iOS apps https://developer.apple.com/documentation/uikit/uiaccessibilitytraits

It seems like it was a decision from the React-Native’s team to emulate the Android's role, and on iOS they're appending custom strings to an element's accessibility hint. There’s no way to accomplish automatic localization here unless the react-native package supplies strings for every language in some way

I’m not sure if this behavior is ideal, maybe we should just use the “button” role plus the “selected” state, both actually exist on UIKit so they’re correctly localized. Also, hints can be disabled on the VoiceOver settings, so all these custom roles will not be announced, while the "button" role, which maps to a valid UIKit accessibility trait, will continue to be announced. RN’s team probably went with their approach for a reason, but I'm still learning about this subject and it would be nice to have the basis for this decision clarified. Is it better to have this custom role, or is using the iOS's native 'button' role better? Like `accessibilityRole={Platform.OS === 'android' ? 'radio' : 'button'}` ? Maybe should we use both the native 'button' role and the custom text as a hint?

At least, I think this seemingly non-standard behavior could be mentioned on the docs with instructions for how to localize it. I think every custom role present in that file will have the same issues https://github.com/facebook/react-native/blob/main/React/AccessibilityResources/en.lproj/Localizable.strings, and the initial Pull Request description seems to imply that the strings were moved to that file to make the override possible.

**Comment by fabriziobertoglio1987** [comment]
Do you still experience this issue? 

I have four years of experience maintaining facebook/react-native and I specialize in the Text and TextInput components. I currently have 58 [facebook/react-native PRs](https://github.com/facebook/react-native/pulls?q=is%3Apr+author%3Afabriziobertoglio1987+). 

If you still experience this issue, I will prepare a patched release with the fix.

Thanks a lot
